### PR TITLE
MGMT-2462 AI on OCP cluster

### PR DESCRIPTION
**Pull request by danielerez**
- Added 'deploy-on-ocp-cluster' target: automates the deployemnt of
  service and UI on the test OCP cluster (using build/kubeconfig by default).
- Added 'ocp.py': for updating /etc/hosts.
- Added 'ocp.sh': for deploying service/ui using assited-service (invoked with 'skipper run').

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -199,6 +200,22 @@ kill_port_forwardings:
 kill_all_port_forwardings:
 	scripts/utils.sh kill_port_forwardings '$(SERVICE_NAME) $(UI_SERVICE_NAME)'
 
+deploy_service_on_cluster:
```
Comment (yevgeny-shnaidman) [inline Makefile:203]: lets rename it to "deploy_service_on_ocp_cluster

File: scripts/day2.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+function deploy_day2_service() {
```
Comment (yevgeny-shnaidman) [inline scripts/day2.sh:4]: let's rename deploy_ocp_service and the file should be ocp.sh

File: discovery-infra/day2.py [hunk 3]
```diff
@@ -215,3 +222,79 @@ def _day2_cluster_create_params(openshift_version, api_vip_dnsname):
         "api_vip_dnsname": api_vip_dnsname,
     }
     return params
+
+def get_ocp_cluster(args):
```
Comment (yevgeny-shnaidman) [inline discovery-infra/day2.py:226]: lets keep day2.py for day2 on clud operations, and move your need code  to a new file ( for example ocp.py

File: Makefile (hunk 1 above)
Comment (danielerez) [reply Makefile:203]: done

File: scripts/day2.sh (hunk 2 above)
Comment (danielerez) [reply scripts/day2.sh:4]: done

File: discovery-infra/day2.py (hunk 3 above)
Comment (danielerez) [reply discovery-infra/day2.py:226]: done

File: Makefile [hunk 4]
```diff
@@ -199,6 +200,22 @@ kill_port_forwardings:
...
+		scripts/deploy_assisted_service.sh
+
+	# UI
+	DEPLOY_TARGET=ocp NAMESPACE_INDEX=$(shell bash scripts/utils.sh get_namespace_index $(NAMESPACE)) \
+		DEPLOY_TAG=$(DEPLOY_TAG) DEPLOY_MANIFEST_TAG=$(DEPLOY_MANIFEST_TAG) SERVICE_KUBECONFIG=$(SERVICE_KUBECONFIG) \
+		CLUSTER_NAME=$(CLUSTER_NAME) CLUSTER_ID=$(CLUSTER_ID) PROFILE=$(PROFILE) \
+		scripts/deploy_ui.sh
+
+config_etc_hosts_for_ocp_cluster:
+	discovery-infra/ocp.py --config-etc-hosts -cn $(CLUSTER_NAME) -id $(CLUSTER_ID) -ns $(NAMESPACE) --service-name $(SERVICE_NAME) --profile $(PROFILE) --deploy-target ocp $(ADDITIONAL_PARAMS)
```
Comment (yevgeny-shnaidman) [inline Makefile:217]: i think we can move it under deploy_service_on_ocp_cluster,no?

Comment (yevgeny-shnaidman) [inline Makefile:217]: do we need all those parameters? Can we use just cluster_id?

Comment (danielerez) [reply Makefile:217]: no, since we use it in ocp.sh, as it's invoked using skipper.

Comment (danielerez) [reply Makefile:217]: need those for get_assisted_service_url_by_args, as we create an assisted-service client for getting the cluster.

File: discovery-infra/ocp.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        '--profile',
+        help='Minikube profile for assisted-installer deployment',
+        type=str,
+        default='assisted-installer'
+    )
+    parser.add_argument(
+        '--deploy-target',
+        help='Where assisted-service is deployed',
+        type=str,
+        default='minikube'
```
Comment (RazRegev) [inline discovery-infra/ocp.py:71]: shouldn't it be 'ocp'?

File: discovery-infra/ocp.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        sys.stdout.write(ocp_cluster.api_vip)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="OCP")
+    parser.add_argument(
+        "-id", "--cluster-id",
+        help="Cluster id to install",
+        type=str,
+        default=None
```
Comment (RazRegev) [inline discovery-infra/ocp.py:40]: don't think you need to explicitly add `default=None `

File: discovery-infra/ocp.py [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    client = assisted_service_api.create_client(
+            url=utils.get_assisted_service_url_by_args(args=args)
+    )
+    return client.cluster_get(cluster_id=args.cluster_id)
+
+def main(args):
+    if args.config_etc_hosts:
+        ocp_cluster = get_ocp_cluster(args)
+        ocp_cluster_name = ocp_cluster.name
+        ocp_api_vip_dnsname = "api." + ocp_cluster_name + "." + ocp_cluster.base_dns_domain
```
Comment (RazRegev) [inline discovery-infra/ocp.py:27]: nit: ` f"api.{ocp_cluster_name}.{ocp_cluster.base_dns_domain}`

File: scripts/deploy_assisted_service.sh [hunk 8]
```diff
@@ -11,13 +11,25 @@ export SERVICE_PORT=$(( 6000 + $NAMESPACE_INDEX ))
 export SERVICE_BASE_URL=${SERVICE_BASE_URL:-"http://${SERVICE_URL}:${SERVICE_PORT}"}
 export EXTERNAL_PORT=${EXTERNAL_PORT:-y}
 export PROFILE=${PROFILE:-assisted-installer}
+export OCP_SERVICE_PORT=$(( 7000 + $NAMESPACE_INDEX ))
```
Comment (RazRegev) [inline scripts/deploy_assisted_service.sh:14]: not sure I understand, why should it run with another  port then the default assisted-service port 6000?
as I understand they are not overriding each other

File: discovery-infra/ocp.py (hunk 5 above)
Comment (danielerez) [reply discovery-infra/ocp.py:71]: good call, better to change the default.

File: discovery-infra/ocp.py (hunk 6 above)
Comment (danielerez) [reply discovery-infra/ocp.py:40]: for sure? copied it from: https://github.com/openshift/assisted-test-infra/blob/a425c840c727b4e3be1c2341e8b70930b10cb216/discovery-infra/install_cluster.py#L158
maybe it's defaulted to an empty string if not specified? I'll give it a try

File: discovery-infra/ocp.py (hunk 7 above)
Comment (danielerez) [reply discovery-infra/ocp.py:27]: done

File: scripts/deploy_assisted_service.sh (hunk 8 above)
Comment (danielerez) [reply scripts/deploy_assisted_service.sh:14]: we should have only one per cluster, but 6000 is already used by the 'parent' service. so in order to reach both with same hostname, we use different ports.

**Comment by danielerez** [comment]
@tsorya PTAL

**Review by RazRegev — APPROVED** [review]

**Pull request merged**